In [1]:
import statsmodels.api as sm
from statsmodels.formula.api import ols
import pandas as pd

In [2]:
data = pd.read_excel('anova_lab_data.xlsx')
data

Power   Etching Rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

## Part 1

### Null Hypothesis

* The null hypothesis will show no difference in the mean of the etching rates for different levels of power of the plasma beam

### Alternate Hypothesis

* The alternate hypothesis tell us there is at least one significant difference in the mean etching rates for the different levels of power

### Significance Level

* The significance level is set at 0.05 (α). This means we would reject the null hypothesis if the p-value is less than 0.05.

### Degrees of freedom

* To calculate the degrees of freedom, we need to know the number of levels of power (different groups) and the total number of observations

The degrees of freedom for the ANOVA are as follows:

1. Degrees of Freedom Between Groups: 
df_between = 2

2. Degrees of Freedom for the Error Within Groups: 
df_within = 12

3. Total Degrees of Freedom: 
df_tota = 14

## Part 2

In [8]:
data.columns = data.columns.str.replace(' ', '_')

In [12]:
model = ols('Etching_Rate ~ C(Power_)',data=data).fit()
table = sm.stats.anova_lm(model)
print(table)

             df     sum_sq   mean_sq          F    PR(>F)
C(Power_)   2.0  18.176653  9.088327  36.878955  0.000008
Residual   12.0   2.957240  0.246437        NaN       NaN


Interpretation:
* The p-value is much smaller than the significance level α = 0.05
* This means we reject the null hypothesis.
* In an F-distribution table, we would compare our F-value (36.878955) to the critical value for F at our significance level (0.05) and degrees of freedom (2 and 12). The F value is greater than the critical value so we reject the null hypothesis

## Conculsion

There is a statistically significant difference in the mean etching rates for the different power levels of the plasma beam.

The significance level is 0.05, which confirms the presence of significant differences among the groups